packages/web-components/fast-foundation/src/menu/stories/menu.register.ts

```diff
@@ -54,39 +56,51 @@ export class FancyMenu extends FASTMenu {
     protected setItems(): void {
         super.setItems();
 
-        this.menuItems
-            ?.filter(this.isMenuItemElement)
-            .forEach((item: HTMLElement, index: number) => {
-                const indent: FancyMenuItemColumnCount = menuItems.reduce(
-                    (accum, current) => {
-                        const elementValue = FASTMenu.elementIndent(current);
-
-                        return Math.max(
-                            accum,
-                            elementValue as number
-                        ) as FancyMenuItemColumnCount;
-                    },
-                    0
-                );
-
-                if (item instanceof FancyMenuItem) {
-                    item.startColumnCount = indent;
+        this.menuItems?.filter(this.isMenuItemElement).forEach((item: HTMLElement) => {
+            const indentInfo: FancyMenuIndentInfo = this.menuItems!.reduce(
+                (accum: FancyMenuIndentInfo, current: HTMLElement) => {
+                    const elementValue = FancyMenu.elementIndent(current);
+
+                    return {
+                        hasControl: accum.hasControl || elementValue.hasControl,
+                        hasStart: accum.hasStart || elementValue.hasStart,
+                    };
+                },
+                {
+                    hasControl: false,
+                    hasStart: false,
                 }
-            });
+            );
+            const indent =
+                indentInfo.hasControl && indentInfo.hasStart
+                    ? 2
+                    : indentInfo.hasControl || indentInfo.hasStart
+                    ? 1
+                    : 0;
+
+            if (item instanceof FancyMenuItem) {
+                item.startColumnCount = indent;
+            }
+        });
     }
 
-    private static elementIndent(el: HTMLElement): FancyMenuItemColumnCount {
+    private static elementIndent(el: HTMLElement): FancyMenuIndentInfo {
         const role = el.getAttribute("role");
         const startSlot = el.querySelector("[slot=start]");
 
-        if (role && role !== MenuItemRole.menuitem) {
-            return !startSlot ? 1 : 2;
-        }
-
-        return !startSlot ? 0 : 1;
+        return {
+            hasControl: role !== MenuItemRole.menuitem,
+            hasStart: startSlot !== null,
+        };
     }
 }
 
+/**
+ * Type for menu item column count info.
+ * @internal
+ */
+type FancyMenuIndentInfo = { hasControl: boolean; hasStart: boolean };
+
```

Comment:
If we're going with 1:1 with what we had prior to this being removed, then this should be unnecessary. Did the structural change necessitate this?